src/views/ClusterViz/index.js

```diff
@@ -0,0 +1,239 @@
+import React, {
+    PureComponent,
+} from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+import BoundError from '#rs/components/General/BoundError';
+import LoadingAnimation from '#rs/components/View/LoadingAnimation';
+import Table from '#rs/components/View/Table';
+import ForceDirectedGraphView from '#rs/components/Visualization/ForceDirectedGraphView';
+
+import {
+    reverseRoute,
+    groupList,
+    compareString,
+    compareNumber,
+} from '#rs/utils/common';
+
+import AppError from '#components/AppError';
+
+import _ts from '#ts';
+
+import {
+    forceDirectedDataSelector,
+    projectClusterDataSelector,
+    projectIdFromRouteSelector,
+    setProjectClusterDataAction,
+} from '#redux';
+import { pathNames } from '#constants/';
+
+import ProjectClusterDataRequest from './requests/ProjectClusterDataRequest';
+import InitProjectClusterRequest from './requests/InitProjectClusterRequest';
+
+import styles from './styles.scss';
+
+const propTypes = {
+    activeProject: PropTypes.number.isRequired,
+    projectClusterData: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
+    setProjectClusterData: PropTypes.func.isRequired,
+    className: PropTypes.string,
+};
+
+const defaultProps = {
+    className: '',
+};
+
+const mapStateToProps = state => ({
+    activeProject: projectIdFromRouteSelector(state),
+    forcedDirectedData: forceDirectedDataSelector(state),
+    projectClusterData: projectClusterDataSelector(state),
+});
+
+const mapDispatchToProps = dispatch => ({
+    setProjectClusterData: params => dispatch(setProjectClusterDataAction(params)),
+});
+
+@BoundError(AppError)
+@connect(mapStateToProps, mapDispatchToProps)
+export default class ClusterViz extends PureComponent {
+    static propTypes = propTypes;
+    static defaultProps = defaultProps;
+
+    state = {
+        createClusterPending: true,
+        clusterDataPending: true,
+    }
+
+    componentWillMount() {
+        this.stopRequestForInitCluster();
+        this.stopRequestForClusterData();
+        this.startRequestForInitCluster();
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const { activeProject } = this.props;
+        if (activeProject !== nextProps.activeProject) {
+            this.stopRequestForInitCluster();
+            this.stopRequestForClusterData();
```

Comment:
Add stopping the request check inside the start process so that you don't have to do it everytime before starting the request